dialects/linux/dproc.c

```diff
@@ -1518,12 +1522,71 @@ process_proc_map(p, s, ss)
 		efs = sv = 1;
 	    else
 		efs = 0;
-	    if (!efs) {
-		if (HasNFS)
-		    sv = statsafely(fp[6], &sb);
-		else
-		    sv = stat(fp[6], &sb);
-	    }
+
+	/*
+	 * Fork a child process and create a pipe.
+	 */
+            if (pipe(pipefd) < 0) {
+                fprintf(stderr, "%s: can't open pipes: %s\n",
+                    Pn, strerror(errno));
+                Exit(1);
```

Comment:
Is this error fatal? Can we fallback to our original code when pipe() fails?

Reply:
Helpful suggestion, lack of consideration, I'll amend it.

Reply:
Got it, thanks.